crypto/evp/evp_enc.c

```diff
@@ -206,7 +206,7 @@ int EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
 
     ctx->cipher = cipher;
     if (ctx->provctx == NULL) {
-        ctx->provctx = ctx->cipher->newctx();
+        ctx->provctx = ctx->cipher->newctx(cipher->prov);
```

Comment:
Shouldn't this be `ossl_provider_ctx(cipher->prov)`

Reply:
Uhmmmm, yes it should.  Apparently, we have nothing using this, or some tests would have crashed.

Reply:
(the difference between `void *` and `void *` is, as usual, not very clear, and leaves space for intricate butgs)

Reply:
Well, seeing as you're replacing no argument with an argument I'm not surprised that nothing is using that argument yet :-)